### ISSUE TITLE
lstsq and memory

### ISSUE THREAD

**Issue by chrisfilo** [poldracklab/fmriprep#1162]
I'm seeing a lot of OpenNeuro jobs failing with this error:

```
[Node] Error on "fmriprep_wf.single_subject_M80325206_wf.func_preproc_task_ritl_acq_MB_run_04_wf.bold_confounds_wf.signals" (/root/src/fmriprep/work/fmriprep_wf/single_subject_M80325206_wf/func_preproc_task_ritl_acq_MB_run_04_wf/bold_confounds_wf/signals)
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 11, in <module>
    load_entry_point('fmriprep==1.0.15', 'console_scripts', 'fmriprep')()
  File "/usr/local/miniconda/lib/python3.6/site-packages/fmriprep/cli/run.py", line 314, in main
    fmriprep_wf.run(**plugin_settings)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/workflows.py", line 595, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 168, in run
    self._clean_queue(jobid, graph, result=result))
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/base.py", line 227, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 68, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 480, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 564, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 644, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base/core.py", line 520, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/nilearn.py", line 97, in _run_interface
    signals.append(masker.fit_transform(self.inputs.in_file))
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/nifti_maps_masker.py", line 213, in fit_transform
    return self.fit().transform(imgs, confounds=confounds)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/base_masker.py", line 176, in transform
    return self.transform_single_imgs(imgs, confounds)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/nifti_maps_masker.py", line 319, in transform_single_imgs
    verbose=self.verbose)
  File "/usr/local/miniconda/lib/python3.6/site-packages/sklearn/externals/joblib/memory.py", line 283, in __call__
    return self.func(*args, **kwargs)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/base_masker.py", line 98, in filter_and_extract
    memory_level=memory_level)(imgs)
  File "/usr/local/miniconda/lib/python3.6/site-packages/sklearn/externals/joblib/memory.py", line 283, in __call__
    return self.func(*args, **kwargs)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/nifti_maps_masker.py", line 29, in __call__
    mask_img=self._resampled_mask_img_)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/regions/signal_extraction.py", line 270, in img_to_signals_maps
    data[maps_mask, :])[0].T
  File "/usr/local/miniconda/lib/python3.6/site-packages/scipy/linalg/basic.py", line 909, in lstsq
    resids = np.sum(np.abs(x[n:])**2, axis=0)
MemoryError
```

I wonder if this has something to do with the use of joblib and nilearns memoization.

**Comment by effigies** [comment]
Is there much penalty on OpenNeuro for having uncompressed BOLD and MNI-space series in the working directory? Because `--low-mem` will help with this.

**Comment by chrisfilo** [comment]
Unfortunately this is happening even with low mem enabled.

**Comment by effigies** [comment]
I see. We disabled detrending, so it's unclear why they're using least squares at all. This is another argument for writing a pure nibabel signal extractor, IMO.

**Comment by oesteban** [comment]
I'll cut a new release as soon as #1160.

Since #1170 is now merged, this problem should be lesser with the new release.

**Comment by oesteban** [comment]
Hey @chrisfilo, can we re-run latest 1.1.3 and see whether the new implementation of signal extraction solves this?

**Comment by chrisfilo** [comment]
Unfortunately this is not possible at this moment.

**Comment by oesteban** [comment]
Okay, how can I try to replicate this on Sherlock? (meaning: some dataset where this happened and specs of the instance memory, cpu).

**Comment by chrisfilo** [comment]
You could artificially limit memory. Alternatively you can try to use the monitor to compare memory consumption before and after the fix.

**Comment by oesteban** [comment]
This should be fixed by #1438. From that point on (https://github.com/poldracklab/fmriprep/pull/1438/files#diff-38dbce629f52845460f7d97b81dc8b02R19) we are using the improved SignalExtraction interface that @effigies wrote. Before, nilearn's was still being used (despite Chris' implementation).

Please reopen if somebody sees this again.
